tests/test_chain.py

```diff
@@ -159,202 +159,254 @@ def test_reorg_repro(revault_network, bitcoind):
-            [
-                "Detected reorg",
-                # 7 because simple_reorg() adds a block
-                f"Vault deposit '{deposit}' still has '7' confirmations",
-            ]
-        )
+    # Sanity check the timestamps
+    for field in timestamps_from_status(vault["status"]):
+        assert vault[field] is not None, field
 
-    # Now actually shift it (7 + 1 - 3 == 5)
-    bitcoind.simple_reorg(vault["blockheight"], shift=3)
+    # Mine a block and reorg it, it should not affect us since the deposit would still
+    # have more than 6 confs.
+    bitcoind.generate_block(1)
+    height = bitcoind.rpc.getblockcount()
+    for w in revault_network.participants():
+        wait_for(lambda: w.rpc.getinfo()["blockheight"] == height)
+    reorg(revault_network, bitcoind, stop_wallets, height)
+    new_tip = f"{height + 1}.*{bitcoind.rpc.getblockhash(height + 1)}"
     for w in revault_network.participants():
         w.wait_for_logs(
             [
                 "Detected reorg",
-                f"Vault deposit '{deposit}' ended up with '5' confirmations",
-                "Rescan of all vaults in db done.",
+                f"Found common ancestor at height {height - 1}",
+                f"Vault deposit '{deposit}' still has {initial_confs} confirmations at common ancestor",
+                "Rescan .*done",
+                f"New tip.* {new_tip}",
             ]
         )
-        wait_for(
-            lambda: len(w.rpc.listvaults(["unconfirmed"], [deposit])["vaults"]) == 1
-        )
-        for field in ["funded_at", "secured_at", "delegated_at", "moved_at"]:
-            assert w.rpc.listvaults()["vaults"][0][field] is None
-
-    # All presigned transactions must have been removed from the db,
-    # if we get it confirmed again, it will re-create the pre-signed
-    # transactions. But they are the very same than previously to the
-    # signatures on the coordinator are still valid therefore the signature
-    # fetcher thread will add them all and the vault will be back to 'secured'
-    # again
-    bitcoind.generate_block(1)
+        v = w.rpc.listvaults([], [deposit])["vaults"][0]
+        assert v["status"] == vault["status"]
+        for field in timestamps_from_status(vault["status"]):
+            assert v[field] is not None, field
     for w in revault_network.participants():
-        w.wait_for_secured_vaults([deposit])
+        wait_for(lambda: w.rpc.getinfo()["blockheight"] == height + 1)
 
-    # TODO: eventually try with tx malleation
+    height = bitcoind.rpc.getblockcount()
+    vault = w.rpc.listvaults([], [deposit])["vaults"][0]
+    confs = height + 1 - vault["blockheight"]
+    logging.info(
+        f"After first reorg. Vault blockheight {vault['blockheight']} ({confs} confs)"
+    )
 
-    # Now do the same dance with the 'active' status
-    revault_network.activate_vault(vault)
-    bitcoind.simple_reorg(vault["blockheight"] + 3)
+    # Now actually shift it out.
```

Comment:
> as "shift it out" doesn't help

Yes probably a bad use of the language of Shakespeare. What is meant here is "shift so far up the chain that it becomes unconfirmed".

> And just by reading the test, I have no idea why it's unconfirmed now, is it because it has 5 confs?

For this i can't help.
```python
reorg(revault_network, bitcoind, stop_wallets, vault["blockheight"], shift=-1)
```
```python
    def simple_reorg(self, height, shift=0):
        """
        Reorganize chain by creating a fork at height={height} and:
            - If shift >=0:
                - re-mine all mempool transactions into {height} + shift
                  (with shift floored at 1)
            - Else:
                - don't re-mine the mempool transactions
        [...]
        """
```

Reply:
> For this i can't help.

Yes you can, you can put there a nice comment explaining what's happening, so I don't have to remember by heart the meaning of `shift`, or open the `reorg` definition every time I want to read the tests. Try with:
```
# shift=-1 won't re-mine the transactions that were reorged out
```